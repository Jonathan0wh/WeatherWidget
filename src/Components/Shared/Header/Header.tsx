import React from 'react';
import { StyleSheet, View, TouchableOpacity } from 'react-native';
import { useNavigation } from '@react-navigation/native';

import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
import { faBars } from '@fortawesome/free-solid-svg-icons';

const Header = () => {
  const navigation: any = useNavigation();

  return (
    <View style={styles.headerContainer}>
      <TouchableOpacity
        style={styles.settingIcon}
        onPress={() => navigation.openDrawer()}
      >
        <FontAwesomeIcon icon={faBars} size={18} />
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  headerContainer: {
    height: 40,
    justifyContent: 'center'
  },
  settingIcon: {
    marginLeft: 10
  }
});

export default Header;
