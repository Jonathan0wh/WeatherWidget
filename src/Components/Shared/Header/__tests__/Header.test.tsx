import React from 'react';
import { shallow } from 'enzyme';

import Header from '../Header';

describe('<Header />', () => {
  it('should render by default', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
