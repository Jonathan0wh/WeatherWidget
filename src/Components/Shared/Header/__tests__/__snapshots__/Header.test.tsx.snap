// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render by default 1`] = `
<View
  style={
    Object {
      "height": 40,
      "justifyContent": "center",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "marginLeft": 10,
      }
    }
  >
    <Component
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f0c9",
            "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
          ],
          "iconName": "bars",
          "prefix": "fas",
        }
      }
      size={18}
    />
  </TouchableOpacity>
</View>
`;
